src/index.tsx

```diff
@@ -48,10 +49,10 @@ async function init(): Promise<void> {
 
   ReactDOM.render(
     <Provider store={store}>
-      <FeatureToggles features={getCurrentActiveFeatures({ initialFeatures })}>
+      <FeatureToggles features={getCurrentActiveFeatureNames({ initialFeatures })}>
         <Feature>
```

Comment:
I can't think of any reason a `Feature` component should be wrapping the `Router` and `Routes`. `Feature` should wrap the components just inside the `Route`

Reply:
This is probably out of the scope to change here though. You should make it an issue and a high priority one to fix it so that `Feature` is not wrapping the `Router` and `Routes like this.

Reply:
OK I think it was more obvious now to me how to do this since we broke up the package. I added it in PR #161, which would have to either be merged into here or merged in to master after we update the package.

Reply:
Umm it might be a little better, but its not the better way ;)

Reply:
From looking at the code, we have a `components/pages` folder, each of those folders supplies a react router route, so there is no need for this `route` you have added here. You should be wrapping the components that get passed to each route inside their own page folder.

Depending on the behavior you should be creating a `featureOr404` or `featureOrRedirect` higher order component.

Then from that making a `withAuthFeatureOr404` or `withAuthFeatureOrRedirect`. Then you take that higher order component and wrap each page component that needs the feature individually.

Hope that sort of points you in the right direction, if you still  need help we can jump on a call.

Reply:
No need to make make it work in that particular fashion(creating all the HOCs), my point was that each component/page that requires the auth feature should be wrapped individually by the auth Feature toggle, inside the route, not outside the route.

Reply:
@kennylavender how would you address the redirect (that depends on the feature toggle)?

I was thinking a saga, but didn't want to ask for such a big change in the revert-revert... PR... maybe we can create a ticket?

BTW even though ultimately I agree with your comments, https://github.com/poetapp/explorer-web/pull/161/files did look like an improvement to me. Maybe not a definite solution, but it was more readable and less place for bugs IMO

Reply:
> maybe we can create a ticket
Yup we should def handle this in a different ticket/issue.

> I was thinking a saga
Saga in my opinion is the best because it keeps the view dumb, but a `Redirect` component isn't a terrible compromise for short term.

A `Redirect` component uses componentDidMount to change window.location, react-router 4 provides a [Redirect Component](https://reacttraining.com/react-router/web/api/Redirect), can't remember if the prev versions do or not. If not its super easy to create.

Here is a more concrete example of using a Redirect Component, Jesse said he could use some further help earlier so I was going to write out some examples anyways. Lets say we have for example a dashboard page that requires a logged in user.

We wrap the dashboard component with a `Feature` component and the if 'auth' feature is enabled, the `Feature` component will render the `Dashboard` else it will render a `Redirect` component and redirect once it mounts.

```js
import { Redirect } from 'react-router-dom
import { Dashboard } from './Dashboard
import { configureFeature } from '@paralleldrive/react-feature-toggles'

const RedirectToLogin = () => <Redirect to="/login" />

export DashBoardPage = () => {
  <Feature name="auth" activeComponent={Dashboard} inactiveComponent={RedirectToLogin}>
}
```

Or we can use `configureFeature` for a bit more reusability

```js
import { Redirect } from 'react-router-dom
import { Dashboard } from './Dashboard
import { configureFeature } from '@paralleldrive/react-feature-toggles'

const RedirectToLogin = () => <Redirect to={{ pathname: '/login' }} />

const featureOrRedirect = configureFeature(RedirectToLogin) // save this in a separate file somewhere for reuse.

const authFeatureOrRedirect = featureOrRedirect('auth') // save this in another separate file somewhere for reuse.

const DashboardPage = authFeatureOrRedirect(Dashboard)
```

If we wanted to go beyond this by using sagas then this is how I handled it in the past. We create an AuthFeatureRedirect component that dispatches a special authFeatureRedirect action, create an authFeatureSaga that listens for that action and redirects as necessary. Completly removing the logic of how and where we redirect to from the view. 

In the past I used react-router-redux, so I would have the authFeatureSaga listen for the authFeatureRedirect action, then map that to the react-router-redux action `push` that redirects for you. The benefits of using react-router-redux for all your redirects is that you can make a listener of its actions, and log or push those actions elsewhere like analytics, all from a single place.

Sorry for another long comment, I am done typing for now :)

Reply:
Thanks @kennylavender for the thorough explanation!

Those suggestions sound great to me. 

I'd personally go with the RedirectComponent if it's easier and faster to implement in this PR and further improve in separate PRs (even if those come immediately after this one), just to simplify code review and not deviate too much from master.

Reply:
@jwicks31 please create an issue, copy any of the above comments you feel will be helpful for it and I will comment there on how to solve the auth issue along with the auth feature toggle.